From: Jan-Ivar Bruaroey <jan-ivar@users.noreply.github.com>
Date: Fri, 9 Feb 2024 16:30:49 -0500
Subject: (tmp-cherry-pick) Revert "VideoStreamEncoder: exclude screencast from
 capture time measurement." (369111bf30)

This reverts commit 03bc3a0fa67e274efb4518da005f4c5b77c607e9.

Reason for revert: measurement complete

Original change's description:
> VideoStreamEncoder: exclude screencast from capture time measurement.
>
> This CL avoids measurement for screencast encoding work. The reason is
> screencast can cling on to and re-encode old video frames for which
> webrtc::VideoFrame::reference_time() is unchanged.
>
> Bug: chromium:1498378
> Change-Id: I5bf79d29ef7f57ddff2622cbb6c3436480bd16ba
> Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/326103
> Commit-Queue: Markus Handell <handellm@webrtc.org>
> Reviewed-by: Harald Alvestrand <hta@webrtc.org>
> Auto-Submit: Markus Handell <handellm@webrtc.org>
> Cr-Commit-Position: refs/heads/main@{#41097}

Bug: chromium:1498378
Change-Id: I42c1a86123eb1d6c7ad7c8981769f5560884a2f5
Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/327025
Bot-Commit: rubber-stamper@appspot.gserviceaccount.com <rubber-stamper@appspot.gserviceaccount.com>
Reviewed-by: Harald Alvestrand <hta@webrtc.org>
Commit-Queue: Markus Handell <handellm@webrtc.org>
Reviewed-by: Mirko Bonadei <mbonadei@webrtc.org>
Cr-Commit-Position: refs/heads/main@{#41133}
---
 video/video_stream_encoder.cc | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/video/video_stream_encoder.cc b/video/video_stream_encoder.cc
index a5711c104a..c9bd604d05 100644
--- a/video/video_stream_encoder.cc
+++ b/video/video_stream_encoder.cc
@@ -2065,11 +2065,8 @@ void VideoStreamEncoder::EncodeVideoFrame(const VideoFrame& video_frame,
 
   frame_encode_metadata_writer_.OnEncodeStarted(out_frame);
 
-  if (encoder_config_.content_type !=
-      VideoEncoderConfig::ContentType::kScreen) {
-    CaptureProcessingDurationMeasurement::AttachToCurrentVoucher(
-        out_frame.reference_time().value_or(clock_->CurrentTime()));
-  }
+  CaptureProcessingDurationMeasurement::AttachToCurrentVoucher(
+      out_frame.reference_time().value_or(clock_->CurrentTime()));
 
   const int32_t encode_status = encoder_->Encode(out_frame, &next_frame_types_);
   was_encode_called_since_last_initialization_ = true;
